stream.go

```diff
@@ -0,0 +1,355 @@
+type StreamUploadFromURLOptions struct {
+	URL string `json:"url"`
+	UploadVideoOptions
+}
+
+// StreamCreateVideoOptions are parameters used when creating a video.
+type StreamCreateVideoOptions struct {
+	MaxDurationSeconds int        `json:"maxDurationSeconds,omitempty"`
+	Expiry             *time.Time `json:"expiry,omitempty"`
+	UploadVideoOptions
+}
+
+// UploadVideoURLWatermark represents UID of an existing watermark.
+type UploadVideoURLWatermark struct {
+	UID string `json:"uid,omitempty"`
+}
+
+// StreamVideoCreate represents parameters returned after creating a video.
+type StreamVideoCreate struct {
+	UploadURL string               `json:"uploadURL,omitempty"`
+	UID       string               `json:"uid,omitempty"`
+	Watermark StreamVideoWatermark `json:"watermark,omitempty"`
+}
+
+// StreamListOptions represents parameters used when listing stream videos.
+type StreamListOptions struct {
+	After         *time.Time `url:"after,omitempty"`
+	Before        *time.Time `url:"before,omitempty"`
+	Creator       string     `url:"creator,omitempty"`
+	IncludeCounts bool       `url:"include_counts,omitempty"`
+	Search        string     `url:"search,omitempty"`
+	Limit         int        `url:"limit,omitempty"`
+	Asc           bool       `url:"asc,omitempty"`
+	Status        string     `url:"status,omitempty"`
+}
+
+// StreamSignedURLOptions represent parameters used when creating a signed URL.
+type StreamSignedURLOptions struct {
+	ID           string            `json:"id,omitempty"`
+	PEM          string            `json:"pem,omitempty"`
+	EXP          int               `json:"exp,omitempty"`
+	NBF          int               `json:"nbf,omitempty"`
+	Downloadable bool              `json:"downloadable,omitempty"`
+	AccessRules  StreamAccessRules `json:"accessRules,omitempty"`
+}
+
+// StreamVideoResponse represents an API response of a stream video.
+type StreamVideoResponse struct {
+	Response
+	Result StreamVideo `json:"result,omitempty"`
+}
+
+// StreamVideoCreateResponse represents an API response of creating a stream video.
+type StreamVideoCreateResponse struct {
+	Response
+	Result StreamVideoCreate `json:"result,omitempty"`
+}
+
+// StreamListResponse represents the API response from a StreamListRequest.
+type StreamListResponse struct {
+	Response
+	Result []StreamVideo `json:"result,omitempty"`
+	Total  string        `json:"total,omitempty"`
+	Range  string        `json:"range,omitempty"`
+}
+
+// StreamSignedURLResponse represents an API response for a signed URL.
+type StreamSignedURLResponse struct {
+	Response
+	Result struct {
+		Token string `json:"token,omitempty"`
+	}
+}
+
+// StreamAccessRules represents the accessRules when creating a signed URL
+//I feel like should be implemented somewhere already but couldn't find it
+// accessRules under https://api.cloudflare.com/#stream-videos-create-a-signed-url-token-for-a-video
+type StreamAccessRules []map[string]interface{}
```

Comment:
I wasn't sure what the best way to handle this was. The API docs don't really give a full picture of all the possible fields that I could find.

Reply:
could we start with what is documented at https://api.cloudflare.com/#stream-videos-create-a-signed-url-token-for-a-video?